pkg/prog/prog.go

```diff
@@ -63,6 +63,9 @@ func newFlagSet(stderr io.Writer, f *Flags) *flag.FlagSet {
 	fs.BoolVar(&f.BuildInfo, "buildinfo", false, "show build info and quit")
 	fs.BoolVar(&f.JSON, "json", false, "show output in JSON. Useful with -buildinfo.")
 
+	// The `-i` option is for compatibility with POSIX shells so that programs, such as the `script`
+	// command, will work when asked to launch an interactive Elvish shell.
+	fs.Bool("i", false, "force interactive mode; currently ignored")
```

Comment:
Can you please explain what are we trying to implement here? What interactive shell?

Reply:
This is a distinction POSIX shells make. https://tldp.org/LDP/abs/html/intandnonint.html is a good introduction to the issue.

Reply:
@supreetsingh10: See issue #1292 for more detail about "what are we trying to implement here?". The phrase "interactive Elvish shell" is currently underspecified but basically means an Elvish shell started with its stdin and stdout attached to a "terminal" (aka "tty").

Reply:
Thank you Mr. @xiaq and Mr. @krader1961 this cleared out a few things.